Loop/Managers/DeviceDataManager.swift

```diff
@@ -1276,7 +1276,8 @@ extension DeviceDataManager: LoopDataManagerDelegate {
             return units
         }
 
-        let rounded = ([0.0] + pumpManager.supportedBolusVolumes).enumerated().min( by: { abs($0.1 - units) < abs($1.1 - units) } )!.1
+        // only allow volumes that are rounded down
+        let rounded = ([0.0] + pumpManager.supportedBolusVolumes).enumerated().min( by: { abs($0.1 - units) < abs($1.1 - units) && $0.1 < units } )!.1
```

Comment:
Is the force unwrap safe here?

Reply:
Also, is enumeration needed here? It doesn't seem like the index is being used at all. Would this work? 👇🏻 

```suggestion
        let rounded = ([0.0] + pumpManager.supportedBolusVolumes).min(by: { abs($0 - units) < abs($1 - units) && $1 < units })!
```

Reply:
yes, I also thought this code was unusual.